### PR TITLE
feat: get OCI image manifest and configuration.

### LINKED ISSUE #518: Update policy-evaluator to expose the oci-distribution method to get the container image config.
As described in the issue kubewarden/kubewarden-controller#749, we need to update policy-evaluator to expose a method from oci-distribution crate to get that the container image [config](https://github.com/krustlet/oci-distribution/blob/4d75d90e175946b78a564a49d4b89dfc83b29344/src/client.rs#L946). 

#### Acceptance criteria
- [ ] Add callback method to allow policies to get the container image config from a given image. Something similar to what is done to get the container image [manifest](https://github.com/kubewarden/policy-evaluator/blob/64dfcff1279384268fddab0b2d42c3643f57a948/src/callback_handler/oci.rs#L35)
- [ ] Add tests

### PR DESCRIPTION
**Pull request by jvanz**
## Description

Adds a new host capability to allow guest to request an OCI image manifest, its digest and configuration.

Fix #518 

## Test

```shell
make test
```

### CODE REVIEW

**Review by viccuad — APPROVED** [review]
LGTM, pending policy-fetcher.

File: Cargo.toml
Comment (fabriziosestito) [inline Cargo.toml:39]: memo: change this before merging

File: tests/integration_test.rs
Comment (fabriziosestito) [inline tests/integration_test.rs:471]: leftover

File: tests/integration_test.rs
Comment (fabriziosestito) [inline tests/integration_test.rs:444]: This block of setup code seems to be a pattern in every test, maybe we could refactor it to the common module

File: tests/integration_test.rs
Comment (fabriziosestito) [inline tests/integration_test.rs:507]: Instead of using `match` we could simply expect/unwrap twice
e.g.

```rust
    let response_raw = rx
        .await
        .expect("cannot receive response")
        .expect("cannot get response");
```
The same applies for the `test_oci_manifest_capability` test in this file, could we fix this one as well?

**Review by fabriziosestito — CHANGES_REQUESTED** [review]

**Review by viccuad — APPROVED** [review]

**Pull request merged**
